tests/test_write.py

```diff
@@ -12,157 +13,181 @@
     l8_224078_20200518_B2,
 )
 
+NODATA = 32768
 
-class TestConfig(unittest.TestCase):
+
+class TestWrite(unittest.TestCase):
     def test_to_netcdf(self):
-        bands = ['blue', 'green', 'red']
+        bands = ["blue", "green", "red"]
         with tempfile.TemporaryDirectory() as tmp:
-            out_path = Path(tmp) / 'test.nc'
+            out_path = Path(tmp) / "test.nc"
             with gw.open(l8_224078_20200518, band_names=bands) as src:
-                (
-                    src.fillna(32768)
-                    .gw.assign_nodata_attrs(32768)
-                    .astype('uint16')
-                    .gw.to_netcdf(filename=out_path, overwrite=True)
-                )
+                try:
+                    (
+                        src.gw.set_nodata(0, 32768)
+                        .astype("uint16")
+                        .gw.to_netcdf(filename=out_path, overwrite=True)
+                    )
+                except Exception as e:
+                    # If any exception is raised, fail the test with a message
+                    self.fail(f"An error occurred during saving: {e}")
 
     def test_to_raster(self):
         with tempfile.TemporaryDirectory() as tmp:
-            out_path = Path(tmp) / 'test.tif'
+            out_path = Path(tmp) / "test.tif"
+
             with gw.open(l8_224078_20200518) as src:
-                (
-                    src.fillna(32768)
-                    .gw.assign_nodata_attrs(32768)
-                    .astype('uint16')
-                    .gw.to_raster(
-                        filename=out_path,
-                        overwrite=True,
-                        tags={'TEST_METADATA': 'TEST_VALUE'},
-                        compress='lzw',
-                        num_workers=2,
-                    )
+                data = src.gw.set_nodata(0, 32768).astype("uint16")
+                data.gw.to_raster(
+                    filename=out_path,
+                    overwrite=True,
+                    tags={"TEST_METADATA": "TEST_VALUE"},
+                    compress="lzw",
+                    num_workers=2,
                 )
+
                 with gw.open(out_path) as tmp_src:
-                    self.assertTrue(src.equals(tmp_src))
-                    self.assertTrue(hasattr(tmp_src, 'TEST_METADATA'))
-                    self.assertEqual(tmp_src.TEST_METADATA, 'TEST_VALUE')
+                    # Compare array values
+                    self.assertTrue(data.equals(tmp_src))
+                    # Compare attributes
+                    self.assertTrue(
+                        data.gw.nodataval == tmp_src.gw.nodataval == NODATA

```

Comment:
@mmann1123 new test to check that the `nodata` value was properly saved and read.